stock_barcodes_serial_source_relocation/models/stock_picking.py

```diff
@@ -0,0 +1,175 @@
+            "exhausted": True,
+            "line_ids": [
+                (
+                    0,
+                    0,
+                    self._prepare_relocation_inventory_line_values(move_line),
+                )
+            ],
+        }
+
+    def button_validate(self):
+        if (
+            self.picking_type_code != "incoming"
+            and not self.picking_type_id.barcode_option_group_id.allow_negative_quant
+        ):
+            for move_line in self.move_line_ids.filtered(
+                lambda x: x.product_id.tracking == "serial"
+                and x.barcode_relocation_scanned
+            ):
+                quants = move_line.lot_id.quant_ids.filtered(
+                    lambda q: float_compare(
+                        q.quantity,
+                        0,
+                        precision_rounding=move_line.product_id.uom_id.rounding,
+                    )
+                    > 0
+                )
+                if len(quants) > 1:
+                    raise ValidationError(
+                        _(
+                            "The S/N %s has quantities in multiple locations"  # modify
+                            % move_line.lot_id.name
+                        )
+                    )
+                qty_available = quants.filtered(
+                    lambda x: x.location_id == move_line.location_id
+                ).quantity
+                if (
+                    float_compare(
+                        move_line.qty_done,
+                        qty_available,
+                        precision_rounding=move_line.product_id.uom_id.rounding,
+                    )
+                    > 0
+                ):
+                    if quants:
+                        warehouse = move_line.location_id.get_warehouse()
+                        if not warehouse:
+                            raise ValidationError(
+                                _(
+                                    "No warehouse found for the location %s"
+                                    % move_line.location_id.name
+                                )
+                            )
+                        picking_type = (
+                            self.env["stock.picking.type"]
+                            .search(
+                                [
+                                    ("warehouse_id", "in", (warehouse.id, False)),
+                                    ("code", "=", "internal"),
+                                    ("is_regularization", "=", True),
+                                ],
+                            )
+                            .sorted(
+                                lambda x: x.warehouse_id.id, reverse=True
+                            )  # provar sense id # test
+                        )
+                        warehouse_ids = picking_type.mapped(
+                            "warehouse_id.id"
+                        )  # veure que retorna, fer diferents proves
```

Comment:
Is this resolved? Please don't leave this comments like these... Unattended and not in English